.github/workflows/validate.yaml

```diff
@@ -69,38 +64,18 @@ jobs:
           key: ${{ matrix.os }}-xdg-v1-${{ hashFiles('**/with-fuseki.sh') }}
           restore-keys: |
             ${{ matrix.os }}-xdg-v1-
-      - name: Install python dependencies
-        shell: bash
-        # Installing tox-gh-actions to get some enhancement to output rendering
-        # in github actions. Eventually we can maybe collapse the tox-envs job
-        # into this one but there are some limitations of tox-gh-actions which
-        # preclude doing that at the moment.
-        run: |
-          python -m pip install tox tox-gh-actions
-      - name: Install system depdendencies for extensive tests
-        if: ${{ matrix.extensive-tests }}
-        shell: bash
-        run: |
-          if [ "${{ matrix.os }}" == "ubuntu-latest" ]
-          then
-              sudo apt-get install -y libdb-dev
-          elif [ "${{ matrix.os }}" == "macos-latest" ]
-          then
-              brew install berkeley-db@4
-              export BERKELEYDB_DIR=$(brew --prefix berkeley-db@4)
-          fi
-      - name: Validate
+      - name: Install Task
+        uses: arduino/setup-task@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I don't think, we need the github secrets here.

Reply:
I have changed the repository settings to assign restrictive permissions to the token:

![image](https://user-images.githubusercontent.com/4481387/166095754-0cd6a443-3e6b-4a9c-9930-aab2c639cdc3.png)

And I have now set the token permission explicitly in the workflow:

https://github.com/aucampia/rdflib/blob/dd1b97a2684eca863fec18b614cad6a0cbe66ddb/.github/workflows/validate.yaml#L16-L17

```yaml
    permissions:
      contents: read
```

The reason for having this though is that without it the following error occurs:

https://github.com/RDFLib/rdflib/runs/6149683404?check_suite_focus=true

```
Run arduino/setup-task@v1
  with:
    version: 3.x
  env:
    FORCE_COLOR: 1
    XDG_CACHE_HOME: /Users/runner/work/rdflib/rdflib/cache
    DEFAULT_PYTHON: 3.7
    pythonLocation: /Users/runner/hostedtoolcache/Python/3.10.4/x64
Error: API rate limit exceeded for 199.19.85.[2](https://github.com/RDFLib/rdflib/runs/6149683404?check_suite_focus=true#step:8:2)2[6](https://github.com/RDFLib/rdflib/runs/6149683404?check_suite_focus=true#step:8:6). (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
```

From the docs:

> repo-token
> (Optional) GitHub access token used for GitHub API requests. Heavy usage of the action can result in workflow run failures caused by rate limiting. GitHub provides a more generous allowance for Authenticated API requests.
> 
> It will be convenient to use [${{ secrets.GITHUB_TOKEN }}](https://docs.github.com/en/actions/reference/authentication-in-a-workflow).

Reply:
Ok, I understand